rest/data/project_settings_test.go

```diff
@@ -561,6 +561,206 @@ func TestSaveProjectSettingsForSection(t *testing.T) {
 	}
 }
 
+func TestPromoteVarsToRepo(t *testing.T) {
+	for name, test := range map[string]func(t *testing.T, ref model.ProjectRef){
+		"SuccessfullyPromotesAllVariables": func(t *testing.T, ref model.ProjectRef) {
+			projectId := "pId"
+			repoId := "rId"
```

Comment:
Since the project ref is already passed into the test, it might be cleaner to just use `ref.Id` and `ref.RepoRefId` rather than re-define the IDs here.